blynklib_cp.py

```diff
@@ -196,7 +198,20 @@ def _get_socket(self):
             self._state = self.CONNECTING
             self._socket = socket.socket()
             self._socket.connect(socket.getaddrinfo(self.server, self.port)[0][4])
-            self._socket.settimeout(self.SOCK_TIMEOUT)
+            if (self.ssl_cert is not None):
+                self.log('Using SSL socket...')
+                import ssl, os
+                sslContext = ssl.create_default_context()
+                sslContext.verify_mode = ssl.CERT_REQUIRED
+                if (self.ssl_cert == "default"):
+                    caFile = os.path.dirname(__file__) + "/certificate/_blynk-cloudcom.crt"
+                else:
+                    caFile = self.ssl_cert
+                sslContext.load_verify_locations(cafile=caFile)
+                self._socket.settimeout(self.SOCK_SSL_TIMEOUT)
+                self._socket = sslContext.wrap_socket(sock=self._socket, server_hostname=self.server)
+            else:
+                self._socket.settimeout(self.SOCK_TIMEOUT)
```

Comment:
probably you can leave settimeout at line 199 . In ssl it will be just overwrited and else block will be not necessary